sbtc/src/deposits.rs

```diff
@@ -349,49 +350,57 @@ impl DepositScriptInputs {
+/// implies that bit (1 << 22) in the locktime, must be zero.
+///
+/// Note that locktimes used as `OP_CSV` inputs in the reclaim script only
+/// use the 16 least significant bits for the value of the locktime. All
+/// other bits in the 32-bit locktime must be zero or the deposit
+/// transaction will fail validation. When we support time-based locktimes,
+/// a user may set bit (1 << 22) to indicate that the locktime value is
+/// time based, as described in BIP-68.
+///
+/// <https://github.com/bitcoin/bips/blob/17c04f9fa1ecae173d6864b65717e13dfc1880af/bip-0068.mediawiki#specification>
+/// <https://github.com/bitcoin/bips/blob/812907c2b00b92ee31e2b638622a4fe14a428aee/bip-0112.mediawiki#summary>
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct ReclaimScriptInputs {
-    /// This is the lock time used for the OP_CSV opcode in the reclaim
+    /// This is the locktime used for the `OP_CSV` opcode in the reclaim
     /// script. It is not allowed to exceed the bounds expected for a
-    /// 5-byte lock-time in bitcoin-core. It is also not allowed to have
+    /// 4-byte locktime in bitcoin-core. It is also not allowed to have
     /// the [`SEQUENCE_LOCKTIME_DISABLE_FLAG`] bit set.
-    lock_time: i64,
+    lock_time: LockTime,
     /// The reclaim script after the `<locked-time> OP_CSV` part of the
     /// script.
     script: ScriptBuf,
 }
 
 impl ReclaimScriptInputs {
     /// Create a new one
-    pub fn try_new(lock_time: i64, script: ScriptBuf) -> Result<Self, Error> {
-        // We can only use numbers that can be expressed as a 5-byte signed
-        // integer, which has a max of `2**39 - 1`. Negative numbers might
-        // be considered non-standard, so we reject them as well.
-        if lock_time > i64::pow(2, 39) - 1 || lock_time < 0 {
-            return Err(Error::InvalidReclaimScriptLockTime(lock_time));
-        }
-
-        // OP_CSV checks can be disabled if the lock-time has the disabled
-        // lock-time bit set to 1. So we disallow such lock times to ensure
+    pub fn try_new(lock_time: u32, script: ScriptBuf) -> Result<Self, Error> {
+        // OP_CSV checks can be disabled if the locktime has the disabled
+        // locktime bit set to 1. So we disallow such locktimes to ensure
         // that the OP_CSV check is always enabled.
         //
         // <https://github.com/bitcoin/bitcoin/blob/v27.1/src/script/interpreter.cpp#L560-L592>
-        if lock_time & SEQUENCE_LOCKTIME_DISABLE_FLAG != 0 {
-            return Err(Error::InvalidReclaimScriptLockTime(lock_time));
+        let lock_time = LockTime::from_consensus(lock_time).map_err(Error::DisabledLockTime)?;
+
+        // For now, we only accept locktimes denominated in bitcoin block
+        // units.
+        if matches!(lock_time, LockTime::Time(_)) {
+            return Err(Error::UnsupportedLockTimeUnits(
+                lock_time.to_consensus_u32(),
+            ));
         }
```

Comment:
The LockTime bitcoin type is an enum that can support both block height and the time interpretation:

```
pub enum LockTime {
    /// A block height lock time value.
    Blocks(Height),
    /// A 512 second time interval value.
    Time(Time),
}
```

What is the rationale for not supporting both? It's not hard to support in #380

Reply:
Requesting changes until I understand why we're choosing to only support one method. I think this lack of support is something that will catch developers off guard when they don't read the fine print of what we do and don't support.

Reply:
> What is the rationale for not supporting both? It's not hard to support in #380

Well, the lack of support in this PR isn't meant to be the end-all. We can merge this PR and then #380 can remove the restriction if it is straightforward to do so. At the same time, I could see a judgement call being made where we forgo the extra work to support it for now and ship without supporting it.



> Requesting changes until I understand why we're choosing to only support one method. I think this lack of support is something that will catch developers off guard when they don't read the fine print of what we do and don't support.

Well, yeah they'll be caught off guard with this if they don't read the docs. But this is not the only place where this is true, much of our deposit protocol is fine print and we are strict about it's adherence. And I'm not against being mindful of the potential pitfalls a developer may face, I'm just weighing that against the time spent on other work that needs to get done as we sprint towards the finish.